clients/venice-client/src/test/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClientTest.java

```diff
@@ -494,10 +510,175 @@ public void testBatchGetWithExceptionFromTransportLayerForOneRoute(boolean useSt
     }
   }
 
+  /**
+   * Condition to test: batchGet API either returns full results or exception, but no partial results.
+   * In this test:
+   * setup: 1 key returns valid value and the other key doesn't return anything.
+   * Behavior: routingLeakedRequestCleanupThresholdMS times out, so returns exception with
+   *           "At least one route did not complete".
+   */
+  @Test(timeOut = TEST_TIMEOUT, expectedExceptions = ExecutionException.class, expectedExceptionsMessageRegExp = ".*VeniceClientException: At least one route did not complete")
+  public void testBatchGetWithTimeoutV1() throws IOException, ExecutionException, InterruptedException {
+    long routingLeakedRequestCleanupThresholdMS = TimeUnit.SECONDS.toMillis(3);
```

Comment:
Since this is an unit test ideally we should be able to control the time and not having to actually wait. That's probably too much effort for this PR but at least we can make the timeout much smaller. 3 seconds is too long and will add unnecessary wait time.

Reply:
Makes sense. Will reduce the timeouts.

Reply:
Fixed this.